Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -2544,6 +2544,120 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Revoked docs queue behind revs", "
+             { channelID }
+            );
+    enum {
+        iPushPull
+    };
+
+    constexpr slice body = "{\"ans*wer\":42}"_sl;
+    alloc_slice bodyWithChannel = SG::addChannelToJSON(body, "channels", {channelID});
+    alloc_slice localPrefix {idPrefix + "local-"};
+    alloc_slice remotePrefix {idPrefix + "remote-"};
+    unsigned docCount = 20;
+
+    // push documents to the push/pull collection
+    for (size_t i = 0; i < _collectionCount; ++i) {
+        for (int d = 1; d <= docCount; ++d) {
+            constexpr size_t bufSize = 80;
+            char docID[bufSize];
+            snprintf(docID, bufSize, "%.*s%d", SPLAT(remotePrefix), d);
+            createFleeceRev(_collections[i], slice(docID), nullslice, bodyWithChannel);
+        }
+    }
+
+    {
+        // Send the docs to remote
+        ReplParams replParams { _collectionSpecs };
+        replParams.setPushPull(kC4OneShot, kC4Disabled);
+        replicate(replParams);
+    }
+
+    deleteAndRecreateDBAndCollections();
+
+    // add 20(docCount) "local-" docs to Push/Pull collection
+    // The 20(docCount) "remote-" docs are already in the remote db
+    for (size_t i = 0; i < _collectionCount; ++i) {
+        for (int d = 1; d <= docCount; ++d) {
+            constexpr size_t bufSize = 80;
+            char docID[bufSize];
+            snprintf(docID, bufSize, "%.*s%d", SPLAT(localPrefix), d);
+            createFleeceRev(_collections[i], slice(docID), nullslice, bodyWithChannel);
+        }
+    }
+
+    // Perform a Push&Pull replication on db
+    // It will push 20 "local-" docs to the remote, and pull "remote-" docs to local db
+    {
+        ReplParams replParams { _collectionSpecs };
+        replParams.collections[iPushPull].push = kC4OneShot;
+        replParams.collections[iPushPull].pull = kC4OneShot;
+        replicate(replParams);
+    }
+
+    auto check = [&]() {
+        for (size_t i = 0; i < _collectionCount; ++i) {
+            c4::ref<C4DocEnumerator> e = c4coll_enumerateAllDocs(_collections[i],
+                                                                 nullptr, ERROR_INFO());
+            unsigned total = 0;
+            unsigned local = 0;
+            unsigned remote = 0;
+            while (c4enum_next(e, ERROR_INFO())) {
+                C4DocumentInfo info;
+                c4enum_getDocumentInfo(e, &info);
+                slice docID_sl {info.docID};
+                total++;
+                if (docID_sl.hasPrefix(localPrefix)) {
+                    local++;
+                }
+                if (docID_sl.hasPrefix(remotePrefix)) {
+                    remote++;
+                }
+            }
+            switch (i) {
+                case iPushPull:
+                    CHECK(total == 2 * docCount);
+                    CHECK(local == docCount);
+                    CHECK(remote == docCount);
+                    break;
+                default:
+                    break;
+            }
+        }
+    };
+    check();
```

Comment:
This is minor but the question that comes to me is that why do we need to do this check in this test. Can we assume that the replication is correct as there are some other tests that validate this?

Also LiteCore has a way to check the number of docs replicated so the number of docs could be used to verify if all docs are replicated correctly?

Reply:
Yes, it's not the object of this test. It's more a sanity check that sets the basis for the final check that no documents are pushed or pulled. I will change it to REQUIRE